src/Vulkan/VMASharp_API_For_Review.md

```diff
@@ -0,0 +1,307 @@
+  }
+  public struct AllocationContext
+  {
+    public int CurrentFrame;
+    public int FrameInUseCount;
+    public long BufferImageGranularity;
+    public long AllocationSize;
+    public long AllocationAlignment;
+    public AllocationStrategyFlags Strategy;
+    public SuballocationType SuballocationType;
+    public bool CanMakeOtherLost;
+    public AllocationContext(
+      int currentFrame,
+      int framesInUse,
+      long bufferImageGranularity,
+      long allocationSize,
+      long allocationAlignment,
+      AllocationStrategyFlags strategy,
+      SuballocationType suballocType,
+      bool canMakeOtherLost);
+  }
+  [Flags]
+  public enum AllocationCreateFlags
+  {
+    DedicatedMemory = 1,
+    NeverAllocate = 2,
+    Mapped = 4,
+    CanBecomeLost = 8,
+    CanMakeOtherLost = 16, // 0x00000010
+    UpperAddress = 64, // 0x00000040
+    DontBind = 128, // 0x00000080
+    WithinBudget = 256, // 0x00000100
+  }
+  public struct AllocationCreateInfo
+  {
+    public AllocationCreateFlags Flags;
+    public AllocationStrategyFlags Strategy;
+    public MemoryUsage Usage;
+    public MemoryPropertyFlags? RequiredFlags;
+    public MemoryPropertyFlags? PreferredFlags;
+    public uint MemoryTypeBits;
+    public VulkanMemoryPool? Pool;
+    public object? UserData;
+    public AllocationCreateInfo(
+      AllocationCreateFlags flags = (AllocationCreateFlags) 0,
+      AllocationStrategyFlags strategy = (AllocationStrategyFlags) 0,
+      MemoryUsage usage = MemoryUsage.Unknown,
+      MemoryPropertyFlags? requiredFlags = 0,
+      MemoryPropertyFlags? preferredFlags = 0,
+      uint memoryTypeBits = 0,
+      VulkanMemoryPool? pool = null,
+      object? userData = null);
+  }
+  public class AllocationException : VulkanResultException
+  {
+    public AllocationException(string message);
+    public AllocationException(string message, Exception? innerException);
+    public AllocationException(Result res);
+    public AllocationException(string message, Result res);
+  }
+  public struct AllocationPoolCreateInfo
+  {
+    public int MemoryTypeIndex;
+    public PoolCreateFlags Flags;
+    public long BlockSize;
+    public int MinBlockCount;
+    public int MaxBlockCount;
+    public int FrameInUseCount;
+    public Func<long, IBlockMetadata>? AllocationAlgorithmCreate;
+    public AllocationPoolCreateInfo(
+      int memoryTypeIndex,
+      PoolCreateFlags flags = (PoolCreateFlags) 0,
+      long blockSize = 0,
+      int minBlockCount = 0,
+      int maxBlockCount = 0,
+      int frameInUseCount = 0,
+      Func<long, IBlockMetadata>? allocationAlgorithemCreate = null);
+  }
+  public struct AllocationRequest
+  {
+    public const long LostAllocationCost = 1048576;
+    public long Offset;
+    public long SumFreeSize;
+    public long SumItemSize;
+    public long ItemsToMakeLostCount;
+    public object Item;
+    public object CustomData;
+    public AllocationRequestType Type;
+    public readonly long CalcCost();
```

Comment:
I guess the cost will be used internally in the allocator itself to decide something internal ?

Reply:
Yes